### PR TITLE
Datacatalog to csv

### LINKED ISSUE #425: Write data catalogue to tables for reporting
### Kind of request

Adding new functionality

### Enhancement Description

For now, when building a model, an instance of the data catalogue, striped for the data used in model building. Often the information, e.g. data source, version etc, are very relevant to the report, and for republishing the data else where. It would be nice if this data catalogue can be written into a table (e.g. csv or excel) for putting into the report. 

### Use case

When building delft3dfm models, I often work with a combi of local and global data. For the local data, I often use the data catalogue also for documentation purposes. For example I would specify What it is, where the data is coming from, how it was preprocessed and which model component is relevant. These information are provided together with the data themselves to the client in the form of report. Therefore, it would be nice to directly convert these info into a readable table for using in the report. 

It can either generate everything that is in the a new key, e.g. report, or it can identify a few predefined keys in the data catalogue. 

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by Tjalling-dejong**
## Issue addressed
Fixes #425 

## Explanation
Added a save_csv option to Model.write_data_catalog.
## Checklist
- [x] Updated tests or added new tests
- [x] Branch is up to date with `main`
- [x] Tests & pre-commit hooks pass
- [x] Updated documentation if needed
- [x] Updated changelog.rst if needed
- [ ] For predefined catalogs: update the catalog version in the file itself, the references in data/predefined_catalogs.yml, and the changelog in data/changelog.rst

## Additional Notes (optional)
Add any additional notes or information that may be helpful.

### CODE REVIEW

**Review by savente93 — APPROVED** [review]
LGTM

File: tests/test_model.py [hunk 1]
```diff
@@ -163,6 +164,10 @@ def test_write_data_catalog(tmpdir):
     model1.data_catalog.get_source(sources[0]).mark_as_used()
     model1.write_data_catalog(append=True)
     assert list(DataCatalog(data_lib_fn).sources.keys()) == sources[:2]
+    # test writing table of datacatalog as csv
+    model.write_data_catalog(used_only=False, save_csv=True)
+    assert isfile(join(model.root, "hydromt_data.csv"))
+    assert not pd.read_csv(join(model.root, "hydromt_data.csv")).empty
```
Comment (DirkEilander) [inline tests/test_model.py:170]: Can you make this check a bit more precise? E.g. test if `sources[0]` is on line 1 or that the number of lines matches the number of sources.

**Review by DirkEilander — COMMENTED** [review]
Generally looks good to me. Just added a small suggestion to make the tests more precise.

**Review by savente93 — APPROVED** [review]
all good!

**Pull request merged**